forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -32,68 +31,78 @@ internal object LDContextKind {
     const val SERVICE = "service"
 }
 
-internal object LDManager {
-    private val LD_MOBILE_KEY = StopgapGlobalState.envConfig.ldMobileKey
-    private var internalVaultType: VaultType? = null
+internal val ALWAYS_BT = 100.0
+internal val ALWAYS_VGS = 0.0
+
+internal fun computeVaultType(trafficPrimaryPercentFlag: Double): VaultType {
+    val randomNum = Math.random() * 100
+    return if (randomNum < trafficPrimaryPercentFlag) VaultType.BT_VAULT_TYPE else VaultType.VGS_VAULT_TYPE
+}
 
+internal object LDManager {
     private var internalLogger: Log = Log.getSilentInstance()
     private var client: LDClient? = null
 
-    internal var vaultType: VaultType?
-        get() = internalVaultType
-
-        // The setter is only exposed for testing purposes. Otherwise, it is entirely internal and
-        // shouldn't be used directly.
-        set(value) {
-            if (value == null) {
-                internalLogger.w("[LaunchDarkly] vaultType is being reset to null. This should only happen while unit testing!")
-                internalVaultType = null
-            } else if (internalVaultType == null) {
-                internalVaultType = value
-            } else {
-                throw Error("vaultType can only be set once!")
-            }
-        }
+    // default to 100% VGS usage in case LD flag retrieval fails
+    private var primaryTrafficPercent = ALWAYS_VGS
```

Comment:
Good question! I was inspired by the point you surfaced in [the Minimize SDK Failures design doc](https://docs.google.com/document/d/1lO5uFpYwkE-AUYKPWBrWtX90mNVYiCeqkGV1CySMK30/edit#bookmark=id.6nweccylo1j4) to refactor:

`internal var vaultType: VaultType?` ➡️ `private var primaryTrafficPercent = ALWAYS_VGS`

To expand on this, `vaultType` was able to be null which required us to do type-coercion [here](https://github.com/teamforage/forage-android-sdk/blob/1048904bea39af315b1e2c566846f5e8c5304be2/forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt#L96). Here's a screenshot showing the IDE complaint

![image](https://github.com/teamforage/forage-android-sdk/assets/12377418/e0430915-4eb2-400c-b3cd-a5233ca210dc)

In other words, I wanted to eliminate the ability to do `LDManager.vaultType = null` to prevent the possibily of `getVaultProvider` ever accidentally return `null`. Using a non-null `primaryTrafficPercent` variable has no risk `getVaultProvider` returning `null` because `primaryTrafficPercent` is always guaranteed to be a Float.

That was my rationale but let me know your thoughts!